packages/core/mesh/network-manager/src/swarm/connection.ts

```diff
@@ -168,15 +173,26 @@ export class Connection {
       this.close(new ProtocolError('protocol stream closed')).catch((err) => this.errors.raise(err));
     });
 
+    const connectedContext = new Context();
+    scheduleTask(
+      connectedContext,
+      async () => {
+        log.warn('timeout waiting for transport to connect, aborting');
+        await this.abort().catch((err) => this.errors.raise(err));
+      },
+      TRANSPORT_CONNECTION_TIMEOUT,
+    );
+
     invariant(!this._transport);
     this._transport = this._transportFactory.createTransport({
       initiator: this.initiator,
       stream: this._protocol.stream,
       sendSignal: async (signal) => this._sendSignal(signal),
     });
 
-    this._transport.connected.once(() => {
+    this._transport.connected.once(async () => {
       this._changeState(ConnectionState.CONNECTED);
+      await connectedContext.dispose();
```

Comment:
will it trigger `abort` for any case other than successful connection?
since we don't abort in 100% cases on error atm, should we dispose this context inside 
```
this._transport.errors.handle((err) => {
```
as well?

Reply:
good catch.

thinking that it'd be better if the more of the connection establishment flow and state tracking in `Peer` should be moved to `Connection` so it's easier to enforce these sort of timeouts/invariants, but will leave it at this simple `CONNECTING` -> `CONNECTED` timeout until we have more insight into the failure modes.